extensions/fairifier-rdf-extension/dtls/fairifier/FileSystemRdfSkeletonImp.java

```diff
@@ -0,0 +1,92 @@
+package org.dtls.fairifier;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class FileSystemRdfSkeletonImp implements RdfSkeletonService{
+    private static final Path SAVELOCATION = Paths.get(System.getProperty("user.home") + "/.local/share/openrefine/");
+    private final static Logger logger = LoggerFactory.getLogger("FileSystemRdfSkeletonImp");
+
+    @Override
+    public List<String> listModels(String fileType) {
+        File folder = SAVELOCATION.toFile();
+        ArrayList<String> out = new ArrayList<String>();
+        File[] listOfFiles = folder.listFiles();
+        for (int i = 0; i < listOfFiles.length; i++) {
+            if (listOfFiles[i].isFile()) {
+                String name = listOfFiles[i].getName().split(".")[0];
+                try {
+                    ObjectMapper mapper = new ObjectMapper();
+                    SkeletonMetadata metadata = mapper.readValue(new File(listOfFiles[i].getParent() + File.separator + name + "metadata.skeleton.json"), SkeletonMetadata.class);
```

Comment:
If we manage to simplify/modernise the file listing, we can probably simplify this as well. At least the creation of the file or path object, but maybe the conversion step as well. Up for discussion once we figure out the file listing.